src/main/java/com/entrevistador/orquestador/infrastructure/adapter/dao/HojaDeVidaBdDao.java

```diff
@@ -0,0 +1,62 @@
+package com.entrevistador.orquestador.infrastructure.adapter.dao;
+
+import com.entrevistador.orquestador.dominio.excepciones.IdHojaDeVidaRagNuloException;
+import com.entrevistador.orquestador.dominio.excepciones.UsernameNoEncontradoException;
+import com.entrevistador.orquestador.dominio.model.dto.HojaDeVidaDto;
+import com.entrevistador.orquestador.dominio.port.HojaDeVidaDao;
+import com.entrevistador.orquestador.infrastructure.adapter.entity.HojaDeVidaEntity;
+import com.entrevistador.orquestador.infrastructure.adapter.repository.HojaDeVidaRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Repository;
+import reactor.core.publisher.Mono;
+
+@Repository
+@RequiredArgsConstructor
+public class HojaDeVidaBdDao implements HojaDeVidaDao {
+    private final HojaDeVidaRepository hojaDeVidaRepository;
+
+    public Mono<Void> guardarHojaDeVida(HojaDeVidaDto hojaDeVidaDto){
+        return this.hojaDeVidaRepository.save(HojaDeVidaEntity.builder()
+                .username(hojaDeVidaDto.getUsername())
+                .idHojaDeVidaRag(hojaDeVidaDto.getIdHojaDeVidaRag())
+                .nombre(hojaDeVidaDto.getNombre())
+                .perfil(hojaDeVidaDto.getPerfil())
+                .seniority(hojaDeVidaDto.getSeniority())
+                .tecnologiasPrincipales(hojaDeVidaDto.getTecnologiasPrincipales())
+                .experienciasLaborales(hojaDeVidaDto.getExperienciasLaborales())
+                .habilidadesTecnicas(hojaDeVidaDto.getHabilidadesTecnicas())
+                .certificaciones(hojaDeVidaDto.getCertificaciones())
+                .proyectos(hojaDeVidaDto.getProyectos())
+                .nivelIngles(hojaDeVidaDto.getNivelIngles())
+                .otrasHabilidades(hojaDeVidaDto.getOtrasHabilidades())
+                .build()).then(Mono.empty());
+    }
+
+    public Mono<String> obtenerIdHojaDeVidaRag(String username){
+        final String usernameError = "Username no encontrado. Username: %s";
+        final String idRagError = "El campo idHojaDeVidaRag del username \"%s\" es nulo";
+        return this.hojaDeVidaRepository.findByUsername(username)
+                .switchIfEmpty(Mono.error(new UsernameNoEncontradoException(String.format(usernameError, username))))
+                .flatMap(entity -> entity.getIdHojaDeVidaRag() == null ?
+                        Mono.error(new IdHojaDeVidaRagNuloException(String.format(idRagError, username)))
+                        :
+                        Mono.just(entity.getIdHojaDeVidaRag()));
```

Comment:
Esta parte se tiene que modificar. Utilizando el método map para obtener el getIdHojaDeVidaRag y luego el método switchIfEmpty para lanzar el Mono.error .

```suggestion
                .map(HojaDeVidaEntity::getIdHojaDeVidaRag)
                .switchIfEmpty(Mono.error(new IdHojaDeVidaRagNuloException(String.format(idRagError, username))));
```